worlds/yugioh06/__init__.py

```diff
@@ -0,0 +1,538 @@
+
+    def apply_base_patch(self, rom):
+        base_patch_location = "/".join((os.path.dirname(self.__file__), "patch.bsdiff4"))
+        with openFile(base_patch_location, "rb") as base_patch:
+            rom_data = bsdiff4.patch(rom.read(), base_patch.read())
+        if self.options.ocg_arts:
+            ocg_patch_location = "/".join((os.path.dirname(self.__file__), "patches/ocg.bsdiff4"))
+            with openFile(ocg_patch_location, "rb") as ocg_patch:
+                rom_data = bsdiff4.patch(rom_data, ocg_patch.read())
+        structure_deck = self.options.structure_deck
+        if structure_deck.current_key == 'none':
+            draft_patch_location = "/".join((os.path.dirname(self.__file__), "patches/draft.bsdiff4"))
+            with openFile(draft_patch_location, "rb") as draft_patch:
+                rom_data = bsdiff4.patch(rom_data, draft_patch.read())
+        rom_data = bytearray(rom_data)
+        return rom_data
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        slot_data: Dict[str, Any] = {
+            "structure_deck": self.options.structure_deck.value,
+            "banlist": self.options.banlist.value,
+            "final_campaign_boss_unlock_condition": self.options.final_campaign_boss_unlock_condition.value,
+            "fourth_tier_5_campaign_boss_unlock_condition": self.options.fourth_tier_5_campaign_boss_unlock_condition.value,
+            "third_tier_5_campaign_boss_unlock_condition": self.options.third_tier_5_campaign_boss_unlock_condition.value,
+            "final_campaign_boss_challenges": self.options.final_campaign_boss_challenges.value,
+            "fourth_tier_5_campaign_boss_challenges": self.options.fourth_tier_5_campaign_boss_challenges.value,
+            "third_tier_5_campaign_boss_challenges": self.options.third_tier_5_campaign_boss_campaign_opponents.value,
+            "final_campaign_boss_campaign_opponents": self.options.final_campaign_boss_campaign_opponents.value,
+            "fourth_tier_5_campaign_boss_campaign_opponents": self.options.fourth_tier_5_campaign_boss_unlock_condition.value,
+            "third_tier_5_campaign_boss_campaign_opponents": self.options.third_tier_5_campaign_boss_campaign_opponents.value,
+            "number_of_challenges": self.options.number_of_challenges.value,
+        }
+
+        for plando_item in self.multiworld.plando_items[self.player]:
+            if plando_item["from_pool"]:
+                items_to_find = set()
+                for item_type in [key for key in ["item", "items"] if key in plando_item]:
+                    for item in plando_item[item_type]:
+                        items_to_find.add(item)
+                for item in items_to_find:
+                    slot_data[item] = []
+                    for item_location in self.multiworld.find_item_locations(item, self.player):
+                        slot_data[item].extend([item_location.name, item_location.player])
```

Comment:
why is this needed?

Reply:
this puts the plando items into slot data.
I copied it from another apworld. I guess your right and nothing is using it.